src/boundaryEventValidation.js

```diff
@@ -0,0 +1,32 @@
+import isValidBoundaryEventTarget from '@/components/nodes/boundaryEvent/validBoundaryEventTargets';
+import { getEmptyBoundaryEventPositionsForShape } from '@/portsUtils';
+
+export function canAddBoundaryEventToTarget(boundaryEventType, dropTarget) {
+  if (!dropTarget || !isValidBoundaryEventTarget(dropTarget.component)) {
+    return false;
+  }
+
+  if (getEmptyBoundaryEventPositionsForShape(dropTarget).length === 0) {
+    return false;
+  }
+
+  if (boundaryEventType === 'processmaker-modeler-boundary-error-event') {
```

Comment:
I inlined the references to the node types as the imports were causing circular dependency errors.